webapp/components/new-test-results-history-grid.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright 2023 The WPT Dashboard Project. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+import { PolymerElement, html } from '../node_modules/@polymer/polymer/polymer-element.js';
+const pageStyle = getComputedStyle(document.body);
+
+const PASS_COLOR = pageStyle.getPropertyValue('--paper-green-300');
+const FAIL_COLOR = pageStyle.getPropertyValue('--paper-red-300');
+const NEUTRAL_COLOR = pageStyle.getPropertyValue('--paper-grey-300');
+const COLOR_MAPPING = {
+  // Passing statuses
+  OK: PASS_COLOR,
+  PASS: PASS_COLOR,
+
+  // Failing statuses
+  CRASHED: FAIL_COLOR,
+  ERROR: FAIL_COLOR,
+  FAIL: FAIL_COLOR,
+  NOTRUN: FAIL_COLOR,
+  PRECONDITION_FAILED: FAIL_COLOR,
+  TIMEOUT: FAIL_COLOR,
+
+  // Neutral statuses
+  MISSING: NEUTRAL_COLOR,
+  SKIPPED: NEUTRAL_COLOR,
+  default: NEUTRAL_COLOR,
+};
+
+class TestResultsGrid extends PolymerElement {
+  static get template() {
+    return html`
+        <style>
+          .chart {
+            height: 15rem;
+          }
+          .chart rect, .chart text {
+            cursor: pointer;
+          }
+          .browser {
+            height: 2rem;
+            margin-bottom: -0.5rem;
+          }
+        </style>
+        <h2>
+          <img class="browser" alt="chrome chrome,dev,experimental,master,taskcluster,user:chromium-wpt-export-bot,prod logo" src="/static/chrome-dev_64x64.png">
+          Chrome
+        </h2>
+        <div class="chart" id="chrome"></div>
+
+        <h2>
+          <img class="browser" alt="edge azure,dev,edge,edgechromium,experimental,master,prod logo" src="/static/edge-dev_64x64.png">
+          Edge
+        </h2>
+        <div class="chart" id="edge"></div>
+
+        <h2>
+          <img class="browser" alt="firefox experimental,firefox,master,nightly,taskcluster,user:chromium-wpt-export-bot,prod logo" src="/static/firefox-nightly_64x64.png">
+          Firefox
+        </h2>
+        <div class="chart" id="firefox"></div>
+
+        <h2>
+          <img class="browser" alt="safari azure,experimental,master,preview,safari,prod logo" src="/static/safari-preview_64x64.png">
+          Safari
+        </h2>
+        <div class="chart" id="safari"></div>
+        `;
+  }
+
+  static get properties() {
+    return {
+      dataTable: Object,
+      runIDs: Array,
+    };
+  }
+
+  static get is() {
+    return 'new-test-results-history-grid';
+  }
+
+  constructor() {
+    super();
+
+    // Get the test history data and then populate the chart
+    Promise.all([
+      this.getTestHistory(),
+      this.loadCharts()
+    ]).then(() => this.updateAllCharts(this.historicalData));
```

Comment:
Great! This should handle both requests correctly and make sure that they're finished before trying to display the charts.